module/zfs/vdev.c

```diff
@@ -1147,9 +1147,12 @@ void
 vdev_open_children(vdev_t *vd)
 {
 	taskq_t *tq;
-	int children = vd->vdev_children;
+	int children;
 	int c;
+    
+    VERIFY3P(NULL, !=, vd);
 
+    children = vd->vdev_children;
```

Comment:
Extra paranoia is good not needed in this case.  All callers of `vdev_open_children()` have already dereferenced `vd` by this point.  Let's leave this as it was.